primitives/derive/src/lib.rs

```diff
@@ -0,0 +1,6 @@
+mod socket;
+
+#[proc_macro]
+pub fn socket(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
```

Comment:
Please, add a doc comment here. Also some doc-tests will be good under `Examples` section